applications/CoSimulationApplication/custom_utilities/feti_dynamic_coupling_utilities.cpp

```diff
@@ -0,0 +1,696 @@
+        const bool is_implicit = (IsOrigin) ? mIsImplicitOrigin : mIsImplicitDestination;
+
+        SizeType domain_dofs = 0;
+        if (is_implicit)
+        {
+            // Implicit - we can use the system matrix size and equation ordering
+            domain_dofs = pK->size1();
+        }
+        else
+        {
+            // Explicit - we use the nodes with mass in the domain and ordering is just the node index in the model part
+            ModelPart& rDomain = (IsOrigin) ? *mpOriginDomain : *mpDestinationDomain;
+            for (auto node_it : rDomain.NodesArray())
+            {
+                const double nodal_mass = node_it->GetValue(NODAL_MASS);
+                if (nodal_mass > numerical_limit)
+                {
+                    node_it->SetValue(EXPLICIT_EQUATION_ID, domain_dofs);
+                    domain_dofs += dim;
+                }
+            }
+        }
+
+        if (rProjector.size1() != interface_nodes.size() * dim ||
+            rProjector.size2() != domain_dofs)
+            rProjector.resize(interface_nodes.size() * dim, domain_dofs, false);
+        rProjector.clear();
+
+        IndexType interface_equation_id;
+        IndexType domain_equation_id;
+
+        for (size_t i = 0; i < interface_nodes.size(); i++)
+        {
+            interface_equation_id = interface_nodes[i]->GetValue(INTERFACE_EQUATION_ID);
+            domain_equation_id = (is_implicit)
+                ? interface_nodes[i]->GetDof(DISPLACEMENT_X).EquationId()
+                : interface_nodes[i]->GetValue(EXPLICIT_EQUATION_ID);
+
+            for (size_t dof_dim = 0; dof_dim < dim; dof_dim++)
+            {
+                rProjector(interface_equation_id*dim + dof_dim, domain_equation_id + dof_dim) = projector_entry;
+            }
+        }
+
+        // Incorporate force mapping matrix into projector if it is the origin
+        // since the lagrangian multipliers are defined on the destination and need
+        // to be mapped back later
+        if (IsOrigin) ApplyMappingMatrixToProjector(rProjector, dim);
+
+        KRATOS_CATCH("")
+    }
+
+
+    void FetiDynamicCouplingUtilities::CalculateCondensationMatrix(
+        CompressedMatrix& rCondensationMatrix,
+        const CompressedMatrix& rOriginUnitResponse, const CompressedMatrix& rDestinationUnitResponse,
+        const CompressedMatrix& rOriginProjector, const CompressedMatrix& rDestinationProjector)
+    {
+        KRATOS_TRY
+
+        const double origin_gamma = mParameters["origin_newmark_gamma"].GetDouble();
+        const double dest_gamma = mParameters["destination_newmark_gamma"].GetDouble();
+
+        const double origin_dt = mpOriginDomain->GetProcessInfo()[DELTA_TIME];
+        const double dest_dt = mpDestinationDomain->GetProcessInfo().GetValue(DELTA_TIME);
+        double origin_kinematic_coefficient = 0.0;
+        double dest_kinematic_coefficient = 0.0;
+
+        if (mEquilibriumVariableString == "ACCELERATION")
```

Comment:
I changed to class enum now for clarity. I don't use the variables themselves because the equilibrium variable is a string in the project parameters, the VARIABLE& member variable would need to be in the initializer list (since its a ref only) and I want to check if the parameters value is valid. I can't assign the member variable outside of the initializer list because the assignment operator for variables has been removed.